src/storage/src/client.rs

```diff
@@ -111,64 +111,18 @@ impl Storage {
     /// # use google_cloud_storage::client::Storage;
     /// async fn example(client: &Storage) -> gax::Result<()> {
     ///     let response = client
-    ///         .insert_object(
-    ///             "projects/_/buckets/my-bucket",
-    ///             "my-object",
-    ///             "the quick brown fox jumped over the lazy dog",
-    ///         )
+    ///         .insert_object()
+    ///         .set_bucket("projects/_/buckets/my-bucket")
+    ///         .set_object("my-object")
+    ///         .set_payload("the quick brown fox jumped over the lazy dog")
+    ///         .send()
     ///         .await?;
     ///     println!("response details={response:?}");
     ///     Ok(())
     /// }
     /// ```
-    pub async fn insert_object<B, O, P>(
-        &self,
-        bucket: B,
-        object: O,
-        payload: P,
-    ) -> crate::Result<Object>
-    where
-        B: Into<String>,
-        O: Into<String>,
-        P: Into<bytes::Bytes>,
-    {
-        let bucket: String = bucket.into();
-        let bucket_id = bucket
-            .as_str()
-            .strip_prefix("projects/_/buckets/")
-            .ok_or_else(|| {
-                Error::other(format!(
-                    "malformed bucket name, it must start with `projects/_/buckets/`: {bucket}"
-                ))
-            })?;
-        let object: String = object.into();
-        let builder = self
-            .inner
-            .client
-            .request(
-                reqwest::Method::POST,
-                format!("{}/upload/storage/v1/b/{bucket_id}/o", &self.inner.endpoint),
-            )
-            .query(&[("uploadType", "media")])
-            .query(&[("name", &object)])
-            .header("content-type", "application/octet-stream")
-            .header(
-                "x-goog-api-client",
-                reqwest::header::HeaderValue::from_static(&self::info::X_GOOG_API_CLIENT_HEADER),
-            );
-
-        let builder = self.inner.apply_auth_headers(builder).await?;
-        let builder = builder.body(payload.into());
-
-        tracing::info!("builder={builder:?}");
-
-        let response = builder.send().await.map_err(Error::io)?;
-        if !response.status().is_success() {
-            return gaxi::http::to_http_error(response).await;
-        }
-        let response = response.json::<v1::Object>().await.map_err(Error::io)?;
-
-        Ok(Object::from(response))
+    pub fn insert_object(&self) -> InsertObject {
```

Comment:
For a future PR: I think we should restore the required arguments for bucket and object, and maybe payload too.